projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-social-media-message/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+
+export default function useSocialMediaMessage() {
+	const { editPost } = useDispatch( editorStore );
+
+	const { message, maxLength } = useSelect(
+		select => ( {
+			message: select( 'jetpack/publicize' ).getShareMessage(),
+			maxLength: select( 'jetpack/publicize' ).getShareMessageMaxLength(),
+		} ),
+		[]
```

Comment:
Similarly I was wondering whether the empty array would prevent the data updating correctly, but it seems to be working fine. That means the toggle problem could be something else.